cxx/distributions/stringcat.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright 2024
+// See LICENSE.txt
+
+#include <algorithm>
+#include <cassert>
+#include "distributions/stringcat.hh"
+
+int StringCat::string_to_index(const std::string& s) const {
+  auto it = std::find(strings.begin(), strings.end(), s);
```

Comment:
WDYT of instead building a `map<string, int>` in the ctor for faster lookup?

Reply:
I thought about this, but my expectation is that most of the time, the number of strings in the class will be small enough that the speed difference will be minimal.  And saving space by not creating a map isn't entirely inconsequential when running on large datasets.  (Keeping in mind that the number of distinct values in a column isn't the same as the number of rows -- I'm saying that the first will probably be small, but the second might be large.  And because we cluster rows and create Distributions per cluster, that drives the number of instances of this class that get instantiated.  In fact, we might want to consider designs where this class doesn't store its own copy of the vector of strings, but that's a pull request for another day.)